src/inference/src/ie_core.cpp

```diff
@@ -2203,14 +2203,21 @@ void Core::register_plugins(const std::string& xmlConfigFile) {
 
 RemoteContext Core::create_context(const std::string& deviceName, const AnyMap& params) {
     OPENVINO_ASSERT(deviceName.find("HETERO") != 0, "HETERO device does not support remote context");
-    OPENVINO_ASSERT(deviceName.find("MULTI") != 0, "MULTI device does not support remote context");
-    OPENVINO_ASSERT(deviceName.find("AUTO") != 0, "AUTO device does not support remote context");
     OPENVINO_ASSERT(deviceName.find("BATCH") != 0, "BATCH device does not support remote context");
-
+    AnyMap updatedParams = params;
+    std::vector<std::shared_ptr<void>> sovec;
+    for (auto& iter : params) {
+        if (iter.second.is<RemoteContext>()) {
+            updatedParams[iter.first] = iter.second.as<RemoteContext>()._impl;
+            for (auto& tmp : iter.second.as<RemoteContext>()._so)
+                sovec.push_back(tmp);
+        }
+    }
     OV_CORE_CALL_STATEMENT({
-        auto parsed = parseDeviceNameIntoConfig(deviceName, flatten_sub_properties(deviceName, params));
+        auto parsed = parseDeviceNameIntoConfig(deviceName, flatten_sub_properties(deviceName, updatedParams));
         auto remoteContext = _impl->GetCPPPluginByName(parsed._deviceName).create_context(parsed._config);
-        return {remoteContext._ptr, {remoteContext._so}};
+        sovec.push_back(remoteContext._so);
```

Comment:
looks like a WA, why multi remote context cannot store all these so? like MultiExecutableNetwork stores references to device's executable networks via SoPtr

Reply:
Hi ilya, executable network is somehow different, we don't pass compiled model from user side all way through core to plugin to multiexecutable network(while for remote context, the hardware contexts are been created outside on user level, thus we don't need to touch various remote context type from  multi side).
I'll check more on this life cycle related use cases and enable tests